tests/resources/ddo_helpers.py

```diff
@@ -148,50 +141,94 @@ def get_registered_ddo_with_access_service(ocean_instance, wallet, provider_uri=
+    erc721_token, erc20_token = deploy_erc721_erc20(
+        ocean_instance.web3, ocean_instance.config, publisher_wallet
     )
+
+    web3 = ocean_instance.web3
+    config = ocean_instance.config
+    data_provider = DataServiceProvider
+    _, metadata, encrypted_files = create_basics(config, web3, data_provider)
+
+    # Set the compute values for compute service
+    compute_values = {
+        "namespace": "ocean-compute",
+        "cpus": 2,
+        "gpus": 4,
+        "gpuType": "NVIDIA Tesla V100 GPU",
+        "memory": "128M",
+        "volumeSize": "2G",
+        "allowRawAlgorithm": False,
+        "allowNetworkAccess": True,
+    }
     compute_service = Service(
-        service_endpoint=DataServiceProvider.get_url(ocean_instance.config),
+        service_id="2",
         service_type=ServiceTypes.CLOUD_COMPUTE,
-        attributes=compute_attributes,
+        service_endpoint=f"{data_provider.get_url(config)}/api/services/compute",
+        data_token=erc20_token.address,
+        files=encrypted_files,
+        timeout=3600,
+        compute_values=compute_values,
     )
 
-    return get_registered_ddo(
-        ocean_instance, metadata, wallet, compute_service, provider_uri=provider_uri
+    return ocean_instance.assets.create(
+        metadata=metadata,
+        publisher_wallet=publisher_wallet,
+        encrypted_files=encrypted_files,
+        services=[compute_service],
+        erc721_address=erc721_token.address,
+        deployed_erc20_tokens=[erc20_token],
+        encrypt_flag=True,
+        compress_flag=True,
     )
 
 
-def get_registered_algorithm_ddo(ocean_instance, wallet, provider_uri=None):
-    metadata = get_sample_algorithm_ddo_dict()["service"][0]["attributes"]
-    metadata["main"]["files"][0]["checksum"] = str(uuid.uuid4())
-    service = get_access_service(
-        ocean_instance, wallet.address, metadata["main"]["dateCreated"], provider_uri
+def get_registered_algorithm_ddo(ocean_instance: Ocean, publisher_wallet: Wallet):
+    erc721_token, erc20_token = deploy_erc721_erc20(
+        ocean_instance.web3, ocean_instance.config, publisher_wallet
     )
-    if "cost" in metadata["main"]:
-        metadata["main"].pop("cost")
-    return get_registered_ddo(
-        ocean_instance, metadata, wallet, service, provider_uri=provider_uri
+
+    web3 = ocean_instance.web3
+    config = ocean_instance.config
+    data_provider = DataServiceProvider
+    _, metadata, encrypted_files = create_basics(config, web3, data_provider)
+
+    # Update metadata to include algorithm info
+    algorithm_values = {
+        "algorithm": {
+            "language": "scala",
+            "format": "docker-image",
+            "version": "0.1",
+            "container": {
+                "entrypoint": "node $ALGO",
+                "image": "node",
+                "tag": "10",
+                "checksum": "test",
+            },
+        }
+    }
+    metadata.update(algorithm_values)
+
+    access_service = Service(
```

Comment:
you can use the option that autodefines the access service, implicitly. By default, if there are no services given, we create the access service. And there's also options to auto-deploy ERCs. They are much simpler to use and you would require less boilerplate here, in compute. I'd like to see as little code as possible related to publishing here, and focus on compute stuff.

In test_ocean_assets.py you can see examples of publishing both regular assets and compute assets, which you can freely reuse. So the publish part just needs to be transfered here, and used accordingly.

Reply:
>you can use the option that autodefines the access service, implicitly. By default, if there are no services given, we create the access service. And there's also options to auto-deploy ERCs. 
@calina-c Are you referring to `test_ocean_assets.create_asset()`?

Reply:
yes